### PR TITLE
Refactor EC2 Metadata (IMDS) code into new typed accessor library

### PR DESCRIPTION
**Pull request by anguslees**
Refactor EC2 Metadata code into a typed wrapper.  The idea is to split out the low-level mechanism (imds.go) from clever caching or other logic (awsutils.go).

Reviewer: This is intended to be a no-op "obviously safe" refactor of metadata lookup code.  If there's something that looks surprising or unclear, I'm happy to simplify it further.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/awsutils/awsutils.go [hunk 1]
```diff
@@ -501,179 +527,81 @@ func (cache *EC2InstanceMetadataCache) refreshVPCIPv4CIDRs(mac string) error {
...
-			// Can this even happen? To be backwards compatible, we will always return 0 here and log an error.
-			log.Errorf("Device number of primary ENI is %d! It was expected to be 0", deviceNumber)
-			return eniID, 0, nil
+	// TODO: return a simpler data structure.
+	ec2ip4s := make([]*ec2.NetworkInterfacePrivateIpAddress, len(imdsIPv4s))
+	for i, ip4 := range imdsIPv4s {
+		ec2ip4s[i] = &ec2.NetworkInterfacePrivateIpAddress{
+			Primary:          aws.Bool(i == 0),
+			PrivateIpAddress: aws.String(ip4.String()),
 		}
```
Comment (mogren) [inline pkg/awsutils/awsutils.go:595]: Neat, thanks.

**Review by mogren — APPROVED** [review]
Thanks @anguslees, great cleanup.

**Comment by mogren** [comment]
e2e tests all pass: https://github.com/aws/amazon-vpc-cni-k8s/runs/1150477131?check_suite_focus=true

File: pkg/awsutils/awsutils.go [hunk 2]
```diff
@@ -18,42 +18,40 @@ import (
...
-	metadataVPCcidrs     = "/vpc-ipv4-cidr-blocks/"
-	metadataDeviceNum    = "/device-number/"
-	metadataInterface    = "/interface-id/"
+	metadataSGs          = "/security-group-ids"
+	metadataSubnetID     = "/subnet-id"
+	metadataVPCcidrs     = "/vpc-ipv4-cidr-blocks"
+	metadataDeviceNum    = "/device-number"
+	metadataInterface    = "/interface-id"
 	metadataSubnetCIDR   = "/subnet-ipv4-cidr-block"
 	metadataIPv4s        = "/local-ipv4s"
```
Comment (mogren) [inline pkg/awsutils/awsutils.go:56]: Btw, seems like these ones are only used in the tests now, maybe we should move them out?

Comment (anguslees) [reply pkg/awsutils/awsutils.go:56]: Oops, yes meant to include that patch too.  Done.

**Review by mogren — APPROVED** [review]

**Review by jayanthvn — COMMENTED** [review]
Thanks @anguslees :)

**Pull request merged**